core/arch/arm/include/kernel/pseudo_ta.h

```diff
@@ -34,12 +34,12 @@
 #include <user_ta_header.h>
 #include <util.h>
 
-#define PTA_MANDATORY_FLAGS	(TA_FLAG_SINGLE_INSTANCE | \
-				TA_FLAG_MULTI_SESSION | \
+#define PTA_MANDATORY_FLAGS	(TA_FLAG_MULTI_SESSION | \
 				TA_FLAG_INSTANCE_KEEP_ALIVE)
 
 #define PTA_ALLOWED_FLAGS	(PTA_MANDATORY_FLAGS | \
-				TA_FLAG_SECURE_DATA_PATH)
+				 TA_FLAG_SINGLE_INSTANCE | \
```

Comment:
What would it mean to enable `TA_FLAG_SINGLE_INSTANCE` in a pTA ?
pTAs are single instance by construction, even if the flag is not set.

Removing `TA_FLAG_SINGLE_INSTANCE` will also disable the effect of `TA_FLAG_MULTI_SESSION` for the pTA which will now accept only 1 active session.

To get multithreaded pTA, maybe we should rather change `tee_ta_try_set_busy()` to not lock the single instance TA mutex on pTAs entry.

Reply:
You're right, `TA_FLAG_SINGLE_INSTANCE` shouldn't be removed.
Perhaps we can define a new flag instead, `TA_FLAG_CONCURRENT`?
This flag would obviously only be allowed for PTAs.